src/main/java/net/coderbot/iris/gl/shader/StandardMacros.java

```diff
@@ -0,0 +1,184 @@
+		if (minor == null || major == null) return null;
+
+		if (minor.length() == 1) {
+			minor = 0 + minor;
+		}
+		if (bugfix.length() == 1) {
+			bugfix = 0 + bugfix;
+		}
+
+		return major + minor + bugfix;
+	}
+
+	/**
+	 * Returns the current OS String
+	 *
+	 * @return the string based on the current OS
+	 * @see <a href="https://github.com/sp614x/optifine/blob/master/OptiFineDoc/doc/shaders.txt#L687-L692">Optifine Doc</a>
+	 */
+	public static String getOsString() {
+		switch (Util.getOperatingSystem()) {
+			case OSX:
+				return "MC_OS_MAC";
+			case LINUX:
+				return "MC_OS_LINUX";
+			case WINDOWS:
+				return "MC_OS_WINDOWS";
+			case SOLARIS: //Note: Optifine doesn't have a macro for Solaris. https://github.com/sp614x/optifine/blob/master/OptiFineDoc/doc/shaders.txt#L689-L692
+			case UNKNOWN:
+			default:
+				return "MC_OS_UNKNOWN";
+		}
+	}
+
+	/**
+	 * Returns the current GL Version using regex
+	 *
+	 * @param name the name of the gl attribute to parse
+	 * @return current gl version stripped of semantic versioning
+	 * @see <a href="https://github.com/sp614x/optifine/blob/master/OptiFineDoc/doc/shaders.txt#L679-L681">Optifine Doc for GL Version</a>
+	 * @see <a href="https://github.com/sp614x/optifine/blob/master/OptiFineDoc/doc/shaders.txt#L683-L685">Optifine Doc for GLSL Version</a>
+	 */
+	public static String getGlVersion(int name) {
+		String info = GL20.glGetString(name);
+
+		Matcher matcher = SEMVER_PATTERN.matcher(Objects.requireNonNull(info));
+
+		if (!matcher.matches()) {
+			return null;
+		}
+
+		String major = group(matcher, "major");
+		String minor = group(matcher, "minor");
+		String bugfix = group(matcher, "bugfix");
+
+		if (bugfix == null) { //if bugfix is not there, it is 0
+			bugfix = "0";
+		}
+
+		return major + minor + bugfix;
+
+	}
+
+	/**
+	 * Expanded version of {@link Matcher#group(String)} that does not throw an exception.
+	 * If the argument is incorrect (normally resulting in an exception), it returns null
+	 *
+	 * @param matcher matcher to check the group by
+	 * @param name    name of the group
+	 * @return the section of the matcher that is a group, or null, if that matcher does not contain said group
+	 */
+	public static String group(Matcher matcher, String name) {
+		try {
+			return matcher.group(name);
+		} catch (IllegalArgumentException | IllegalStateException exception) {
+			return null;
+		}
+	}
+
+	/**
+	 * Returns the list of currently enabled GL extensions
+	 * This is done by calling {@link GL11#glGetString} with the arg {@link GL11#GL_EXTENSIONS}
+	 *
+	 * @see <a href="https://github.com/sp614x/optifine/blob/master/OptiFineDoc/doc/shaders.txt#L713-L716">Optifine Doc</a>
+	 * @return list of activated extensions prefixed with "MC_"
+	 */
+	public static List<String> getGlExtensions() {
+		String[] extensions = Objects.requireNonNull(GL11.glGetString(GL11.GL_EXTENSIONS)).split("\\s+");
+
+		return Arrays.stream(extensions).map(s -> "MC_" + s).collect(Collectors.toList());
+	}
```

Comment:
It doesn't seem like this detail is supported:

> Only the macros which are referenced and supported are added to the shader file.

While this is fine, that should probably be noted in the code somewhere.